cis194/week3/daniel-deng/src/Exercise01.hs

```diff
@@ -0,0 +1,7 @@
+module CIS194.Homework03.Exercise01 (skips) where
+
+takeEvery :: Int -> [a] -> [a]
+takeEvery n = fmap snd . filter ((== n) . fst) . zip (cycle [1..n])
+
+skips :: [a] -> [[a]]
+skips = fmap . flip takeEvery <*> enumFromTo 1 . length
```

Comment:
`enumFromTo 1` === `range . (1,)` with `import Data.Ix`

Reply:
> `enumFromTo 1` === `range . (1,)` with `import Data.Ix`

Very cool! Thanks for pointing it out @zehua .

Reply:
flipping takeEvery to partially apply it to the list first is a good idea. it avoids the need to generate a repeated list of the input list.

If we aim for shortening the code, we could define takeEvery as the flipped version to get rid of the flip call (4 chars) by adding two `l`s in the definition, giving us a gain of two fewet chars.  :D